src/PythonMigrationViewExtension/GraphPythonDependencies.cs

```diff
@@ -18,10 +23,75 @@ internal GraphPythonDependencies(ViewLoadedParams viewLoadedParams)
         internal bool ContainsIronPythonDependencies()
         {
             var workspace = ViewLoaded.CurrentWorkspaceModel;
+
             if (workspace == null)
                 throw new ArgumentNullException(nameof(workspace));
 
-            return workspace.Nodes.Any(n => IsIronPythonNode(n));
+            if (workspace.Nodes.Any(n => IsIronPythonNode(n)))
+            {
+                return true;
+            }
+
+            // Check if any of the custom nodes has IronPython dependencies in it. 
+            var customNodeManager = ViewLoaded.StartupParams.CustomNodeManager;
+            var customNodes = workspace.Nodes.OfType<Function>();
+
+            return CustomNodeContainsIronPythonDependency(customNodes, customNodeManager);           
+        }
+
+        // This function returns true, if any of the custom nodes in the input list has a IronPython dependency. 
+        // It traverses all CN's in the given list of customNodes and their subtrees until the first CN with an IronPython dependency is found. 
+        // It marks them as containing ironPython dependencies if any of their child custom nodes contain IronPython.
+        internal static bool CustomNodeContainsIronPythonDependency(IEnumerable<Function> customNodes, ICustomNodeManager customNodeManager)
+        {
+            ICustomNodeWorkspaceModel customNodeWS;
+
+            foreach (var customNode in customNodes)
+            {
+                customNodeManager.TryGetFunctionWorkspace(customNode.FunctionSignature, false, out customNodeWS);
+
+                // If a custom node workspace is already checked for Python dependencies, 
+                // check the CustomNodePythonDependency dictionary instead of processing it again. 
+                if (CustomNodePythonDependency.ContainsKey(customNodeWS.CustomNodeId))
+                {
+                    if (CustomNodePythonDependency[customNodeWS.CustomNodeId])
+                    {
+                        return true;
+                    }
+                    continue;
+                }
+
+                var hasPythonNodesInCustomNodeWorkspace = customNodeWS.Nodes.Any(n => IsIronPythonNode(n));
+                CustomNodePythonDependency.Add(customNodeWS.CustomNodeId, hasPythonNodesInCustomNodeWorkspace);
+
+                if (hasPythonNodesInCustomNodeWorkspace)
+                {
+                    return true;
+                }
+
+                // Recursively check for IronPython dependencies in the nested custom nodes.
+                var nestedCustomNodes = customNodeWS.Nodes.OfType<Function>();
+
+                if (nestedCustomNodes.Any())
+                {
+                    hasPythonNodesInCustomNodeWorkspace = CustomNodeContainsIronPythonDependency(nestedCustomNodes, customNodeManager);
+
```

Comment:
Then simply add to the dictionary here:
`CustomNodePythonDependency.Add(customNodeWS.CustomNodeId, hasPythonNodesInCustomNodeWorkspace);`
Then you won't need any of the logic below.